src/main/java/com/aws/greengrass/config/Topic.java

```diff
@@ -325,6 +325,21 @@ public boolean equals(Object o) {
         return false;
     }
 
+    /**
+     * Checks if the Topic has the same name and value.
+     *
+     * @param t the Topic to compare values
+     */
+    public boolean compareValue(Topic t) {
+        if (!getName().equals(t.getName())) {
```

Comment:
My understanding is that it can't be null based off of the existing [equals() function](https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/82039a1d8a1fe81d215e2578614d5d621c7911f2/src/main/java/com/aws/greengrass/config/Topic.java#L323)

Also I will update this line to call equals since it effectively does the same thing